libraries/esp8266/examples/irammem/irammem.ino

```diff
@@ -14,6 +14,204 @@
 #define ETS_PRINTF ets_uart_printf
 #endif
 
+/*
+  Verify mmu_get_uint16()'s compliance with strict-aliasing rules under
+  different optimizations.
+*/
```

Comment:
It's already here, but would this and the rest of the file make sense as a device test file at `/tests/device/test_irammem/` like there's already a umm malloc one? And given the structure of the things below with success and failures

Reply:
I don't know much about how the optimizations might vary across processor vendors, my thinking was this way it is tested with a matching vendor/version of the ESP8266 GCC.

Reply:
The basic idea is to have a separate set of 'tests' that have a clearly defined success and failures, which is useful to have grouped together instead of remembering to run the example code to sanity-check certain Core implementation details. 'Device' is just the board running the test program, and... we are fighting the c++ object lifetime assumptions done by the compiler, not the machine code specifics?

Just to note, idk if it actually plays correctly with something supposed to be hw-only and not mixed with MOCK tests